src/main/java/wooteco/subway/section/domain/Sections.java

```diff
@@ -0,0 +1,138 @@
+                .filter(section -> section.isRelated(requestedSection))
+                .collect(Collectors.toList());
+    }
+
+    private void validateAllDifferent(Section requestedSection, List<Section> relatedSections) {
+        for (final Section section : relatedSections) {
+            validateDifferent(section, requestedSection);
+        }
+    }
+
+    private void validateDifferent(Section firstSection, Section secondSection) {
+        if (firstSection.hasSameStations(secondSection)) {
+            throw new RequestException("이미 두 역을 연결하는 구간이 있습니다.");
+        }
+    }
+
+    private void validateRelated(final Section requestedSection, final List<Section> relatedSections) {
+        if (relatedSections.size() == 2) {
+            final Section firstSection = relatedSections.get(FIRST);
+            final Section secondSection = relatedSections.get(SECOND);
+
+            validateConnected(firstSection, secondSection);
+            validateNoFork(requestedSection, firstSection, secondSection);
+        }
+    }
+
+    private void validateConnected(final Section firstSection, final Section secondSection) {
+        if (connectedFarAway(firstSection, secondSection)) {
+            throw new RequestException("이미 노선에서 서로 연결된 역들입니다.");
+        }
+    }
+
+    private boolean connectedFarAway(final Section firstSection, final Section secondSection) {
+        return !firstSection.canConnect(secondSection);
+    }
+
+    private void validateNoFork(final Section requestedSection, final Section firstSection, final Section secondSection) {
+        if (isForked(requestedSection, firstSection, secondSection)) {
+            throw new RequestException("갈래길을 형성할 수 없습니다.");
+        }
+    }
+
+    private boolean isForked(final Section requestedSection, final Section firstSection, final Section secondSection) {
+        return (requestedSection.canJoin(firstSection) && requestedSection.canJoin(secondSection))
+                || (requestedSection.canConnect(firstSection) && requestedSection.canConnect(secondSection));
+    }
+
+    public List<Section> removeStation(final Long stationId) {
+        validateSectionsLength();
+
+        final List<Section> relatedSections = relatedSectionsOf(stationId);
+        relatedSections.forEach(sections::remove);
+        return Collections.unmodifiableList(relatedSections);
+    }
+
+    private void validateSectionsLength() {
+        if (sections.size() == 1) {
+            throw new RequestException("구간이 하나밖에 없는 노선에서는 역을 제거할 수 없습니다.");
+        }
+    }
+
+    private List<Section> relatedSectionsOf(final Long stationId) {
+        return sections.stream()
+                .filter(section -> section.contains(stationId))
+                .collect(Collectors.toList());
+    }
+
+    public List<Long> toOrderedStationIds() {
+        final Map<Long, Long> stationIdMap = createIdMap(sections);
+
+        final Long beginningUpStation = findBeginningUpStation(stationIdMap);
```

Comment:
개인적으로는 이 부분은 메소드에서 사용할 변수들을 선언 하는 부분이라고 생각해서 공백으로 구분 안하면 좋을 것 같아요 :) 
```suggestion
        final Map<Long, Long> stationIdMap = createIdMap(sections);
        final Long beginningUpStation = findBeginningUpStation(stationIdMap);
```